src/compiler/binder.ts

```diff
@@ -1597,32 +1597,30 @@ namespace ts {
                 if (hasModifier(node, ModifierFlags.Export)) {
                     errorOnFirstToken(node, Diagnostics.export_modifier_cannot_be_applied_to_ambient_modules_and_module_augmentations_since_they_are_always_visible);
                 }
+                const { name } = node;

```

Comment:
u can move this to the else block

Reply:
Control flow was confused because `node` was already an `AmbientModuleDeclaration` and `isExternalModuleAugmentation` re-checks that (thus node was `never` in the  `else`). Broke it into two functions to avoid that.